app/models/category_list.rb

```diff
@@ -64,17 +64,18 @@ def find_relevant_topics
 
     @all_topics = Topic.where(id: category_featured_topics.map(&:topic_id)).includes(:shared_draft, :category)
 
+    @all_topics = @all_topics.includes(:tags).where(tags: {name: @options[:tag]}) if @options[:tag].present?
+
     if @guardian.authenticated?
       @all_topics = @all_topics
         .joins("LEFT JOIN topic_users tu ON topics.id = tu.topic_id AND tu.user_id = #{@guardian.user.id.to_i}")
         .where('COALESCE(tu.notification_level,1) > :muted', muted: TopicUser.notification_levels[:muted])
     end
 
-    @all_topics = TopicQuery.remove_muted_tags(@all_topics, @guardian.user)
-    @all_topics = @all_topics.includes(:last_poster) if @options[:include_topics]
+    @all_topics = TopicQuery.remove_muted_tags(@all_topics, @guardian.user).includes(:last_poster)
     @all_topics.each do |t|
       # hint for the serializer
-      t.include_last_poster = true if @options[:include_topics]
```

Comment:
This is always going to be true 
https://github.com/discourse/discourse/pull/17953/files#diff-c6b98519a4d63acddf8e40a3208c696d289da72c8d177b7a9e685f51b2cad073R16

Reply:
Hmm, I'm not sure I see why `include_topics` is always going to be `true`... can you explain a bit?

Reply:
To clarify a bit, we have a site setting that lets admins choose a `/categories` page layout without topics: 

<img width="360" alt="image" src="https://user-images.githubusercontent.com/368961/184954631-c6f79598-c636-40a0-a7e0-c812cef96ce2.png">

Reply:
> find_relevant_topics if options[:include_topics]

We only call this method if `:include_topics` is true from what I see https://github.com/discourse/discourse/blob/main/app/models/category_list.rb#L16

Reply:
Oh, right, that makes sense. All good to remove the extra checks then, 👍